templates/tftmpl/tfvars.go

```diff
@@ -79,6 +79,26 @@ func newProvidersTFVars(w io.Writer, filename string, input *RootModuleInputData
 	return err
 }
 
+// newVariablesTFVars writes input variables for configured variables.
+func newVariablesTFVars(w io.Writer, filename string, input *RootModuleInputData) error {
```

Comment:
A full on e2e isn't required, but unit tests should be added. Here are some [existing unit tests with golden files that could be used as reference](https://github.com/hashicorp/consul-terraform-sync/blob/64f9d1d9f9990617bfb05c3000f46de0388287fa/templates/tftmpl/golden_test.go#L501 )